codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/generator/awssdk/v2/ToNativeAwsV2.java

```diff
@@ -511,4 +512,66 @@ protected MethodSpec modeledUnion(final UnionShape shape) {
     method.addStatement("return $L.build()", VAR_BUILDER);
     return method.build();
   }
+
+  protected MethodSpec errorOpaque() {
+    final String methodName = "Error";
+    final TypeName inputType = subject.dafnyNameResolver.classForOpaqueError();
+    final ClassName returnType = ClassName.get(RuntimeException.class);
+    return initializeMethodSpec(methodName, inputType, returnType)
+      // If obj is an instance of the Service's Base Exception
+      .beginControlFlow(
+        "if ($L.$L instanceof $T)",
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("obj"),
+        subject.nativeNameResolver.baseErrorForService()
+      )
+      .addStatement(
+        "return ($T) $L.$L",
+        subject.nativeNameResolver.baseErrorForService(),
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("obj")
+      )
+      // If obj is ANY Exception
+      .nextControlFlow(
+        "else if ($L.$L instanceof $T)",
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("obj"),
+        Exception.class
+      )
+      .addStatement(
+        "return ($T) $L.$L",
+        RuntimeException.class,
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("obj")
+      )
+      // If String is set
+      .nextControlFlow(
+        "else if ($L.$L.$L())",
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("message"),
+        Dafny.datatypeConstructorIs("Some")
+      )
+      .addStatement(
+        "final $T message = $L($L.$L.$L)",
```

Comment:
Perhaps this should stringify the opaque error value as well, like the last case below.

Reply:
Done.